.github/workflows/update-package-json.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Check if a version argument is provided
+if [ $# -eq 0 ]; then
+    echo "Please provide a version number as an argument."
+    echo "Usage: $0 <version_number>"
+    exit 1
+fi
+
+# Store the new version
+NEW_VERSION="$1"
+
+# Detect OS and set the appropriate sed command
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    # macOS
+    if command -v gsed >/dev/null 2>&1; then
+        SED_CMD="gsed"
+    else
+        echo "Error: gsed is not installed. Please install it using Homebrew: brew install gnu-sed"
+        exit 1
+    fi
+else
+    # Linux and other Unix-like systems
+    SED_CMD="sed"
+fi
+
+# Find all package.json files in subdirectories
+find . -name "package.json" | while read -r file; do
+    # Check if the file contains any "@dfinity/" package
+    if grep -q '"@dfinity/' "$file"; then
```

Comment:
the only problem I see is that we assume all `@dfinity/` packages to be pinned to the same version. defining a list of known packages in this script and matching against that list might be a better approach

Reply:
From the [agent-js repository](https://github.com/dfinity/agent-js/blob/main/package.json), this is the current list of relevant packages:
* `@dfinity/principal`
* `@dfinity/candid`
* `@dfinity/agent`
* `@dfinity/identity`
* `@dfinity/auth-client`
* `@dfinity/assets`
* `@dfinity/identity-secp256k1`
* `@dfinity/use-auth-client`

Reply:
Right. I think the current assumption from agent-js is that `@dfinity/` all share the same version. Having a list of known packages would be more generic.